### PR TITLE
[Avatar][Card] New components

### LINKED ISSUE #491: Avatar, CardStatus
As a user, I would like additional minor components that enhance the content like status and avatar.

**AC**
* Create CardStatus
* Create Avatar
* Write documentation and review with Wendy for consistency in our site

**Design Reference**
https://ca.invisionapp.com/d/main#/console/11070131/250301791/preview

**Consider:**
* should these live inside Card or should they be separate components?
```
1. Typography
2. Color
3. Spacing
4. Icons
5. Animation
6. States
7. Size / Shape
8. Variants (kinds)
9. A11y
10. Development Docs (props, code, examples)
11. Design Docs (usage, philosophy, patterns)
```

### PR DESCRIPTION
**Pull request by kylegach**
### Description
- Add Avatar
- Add CardStatus
- Add DangerSimple, SuccessSimple, and WarningSimple to bundled Icons
- Add/update/remove demo examples
- Add/update/remove snapshots
- Replace avatar placeholder image with a better one
- Add getColor & getReadableTextColor utilities to the colors directory
    - Quantify the readable color for each color in our palette

### Motivation and context

Closes #491 

### Screenshots, videos, or demo, if appropriate

https://491-card-supplementals--mineral-ui.netlify.com/components/avatar
https://491-card-supplementals--mineral-ui.netlify.com/components/card-status

### How to test

1. `cd mineral-ui && npm start`
2. Navigate to Avatar and CardStatus
3. Confirm things look right

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change

### How does this PR make you feel?

Avatar

### CODE REVIEW

File: src/website/app/demos/Avatar/examples/customTextAbbr.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+/* @flow */
+import Avatar from '../../../../../Avatar';
+
+export default {
+  id: 'custom-text-abbr',
+  title: 'Custom Text Abbreviation',
+  description: `By default, the first letter of text \`children\` will be used
+as the rendered abbreviated text. Pass a string to the \`abb\` prop to provide
```
Comment (coldpour) [inline src/website/app/demos/Avatar/examples/customTextAbbr.js]: abb -> abbr

File: src/Avatar/Avatar.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    | 'lime'
+    | 'magenta'
+    | 'orange'
+    | 'purple'
+    | 'red'
+    | 'sky'
+    | 'slate'
+    | 'teal'
+    | 'theme'
+    | 'yellow',
```
Comment (coldpour) [inline src/Avatar/Avatar.js]: I wish there was a way to have this list in one place. is there any way to pull this from [Color/index.js](https://github.com/mineral-ui/mineral-ui/blob/491-card-supplementals/src/website/app/pages/Color/index.js#L38)?

File: src/website/app/demos/Avatar/examples/avatar.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+/* @flow */
+import Avatar from '../../../../../Avatar';
+import IconCloud from 'mineral-ui-icons/IconCloud';
+import DemoLayout from '../components/DemoLayout';
+
+export default {
+  id: 'basic',
+  title: 'Basic Usage',
+  description: `Avatar can display an image, a text character, or an
+[Icon](../icon). For accessibility, please provide an \`alt\`/\`title\`
```
Comment (coldpour) [inline src/website/app/demos/Avatar/examples/avatar.js]: should alt/title be a required prop?

File: src/website/app/demos/Avatar/bestPractices.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      <DemoLayout>
+        <Avatar>
+          <IconCloud title="cloud" />
+        </Avatar>
+      </DemoLayout>
+    )
+  },
+  {
+    type: 'dont',
+    description: `To prevent duplicate information, hide Avatar from AT with
```
Comment (coldpour) [inline src/website/app/demos/Avatar/bestPractices.js]: what's an "AT?"

File: src/website/app/demos/Card/examples/card-status/importSyntax.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+export default {
+  id: 'import-syntax',
+  title: 'Import Syntax',
+  description: `\`\`\`
+import Card, { CardDivider } from 'mineral-ui/Card';
```
Comment (coldpour) [inline src/website/app/demos/Card/examples/card-status/importSyntax.js]: CardStatus?

File: src/website/app/demos/Avatar/bestPractices.js (hunk 4 above)
Comment (kylegach) [reply src/website/app/demos/Avatar/bestPractices.js]: [Assistive Technology](https://en.wikipedia.org/wiki/Assistive_technology). I'll update to spell it out first.

File: src/website/app/demos/Avatar/examples/avatar.js (hunk 3 above)
Comment (kylegach) [reply src/website/app/demos/Avatar/examples/avatar.js]: The `alt`/`title` is on the `img`/Icon, so Avatar can't enforce that with the current API. `img` should always have an `alt`, even if simply empty, but Icon can have legitimate reason to not have a `title` (like when the title info would be redundant with the icon's text label).

File: src/Avatar/Avatar.js (hunk 2 above)
Comment (brentertz) [reply src/Avatar/Avatar.js]: @coldpour It is possible to import/export Flow types, but we haven't adopted that pattern yet in Mineral.  We started off this project with a very light touch in regards to Flow, after the painful experience with CAAC.  We'll get there before long I'm sure...

File: src/Avatar/Avatar.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+  Avatar_size_medium: baseTheme.size_medium,
+  Avatar_size_large: baseTheme.size_large,
+  Avatar_size_jumbo: baseTheme.size_jumbo,
+
+  ...baseTheme
+});
+
+const Root = createStyledComponent(
+  'span',
+  props => {
```
Comment (brentertz) [inline src/Avatar/Avatar.js]: Suggest to destructure these props...

File: src/Avatar/Avatar.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+      '& > [role="img"]': {
+        fill: color
+      }
+    };
+  },
+  {
+    displayName: 'Avatar',
+    includeStyleReset: true,
+    filterProps: ['icon']
```
Comment (brentertz) [inline src/Avatar/Avatar.js]: Is `filterProps` still needed if you are only using `children` now?

File: src/Avatar/Avatar.js [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    | 'indigo'
+    | 'lime'
+    | 'magenta'
+    | 'orange'
+    | 'purple'
+    | 'red'
+    | 'sky'
+    | 'slate'
+    | 'teal'
+    | 'theme'
```
Comment (brentertz) [inline src/Avatar/Avatar.js]: Suggest to not have a "theme" color/textColor and just have that be the default when color prop is not provided.

File: src/Avatar/Avatar.js [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+);
+
+/**
+ * Avatar provides a graphic representation of an identity. It can display an
+ * image, text, or an [Icon](../icon).
+ */
+export default function Avatar({
+  abbr,
+  children,
+  color = 'theme',
```
Comment (brentertz) [inline src/Avatar/Avatar.js]: Again, suggest to consider not having a default color prop value.

File: src/Avatar/Avatar.js [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+  size = 'large',
+  ...restProps
+}: Props) {
+  let icon, text;
+
+  Children.map(children, child => {
+    if (typeof child === 'string') {
+      text =
+        abbr || child.length > 1 ? (
+          <abbr title={child}>{abbr || child.split('')[0]}</abbr>
```
Comment (brentertz) [inline src/Avatar/Avatar.js]: Could use `child.charAt()` instead of splitting here.

File: src/website/app/demos/Avatar/examples/colors.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+export default {
+  id: 'colors',
+  title: 'Colors',
+  description: `Avatar is available in any of the
+[palette's base colors](/color/#guidelines-ramps). The \`"theme"\` value, which
+is default, will automatically use your theme's base color.`,
+  scope: { Avatar, DemoLayout },
+  source: `
+    <DemoLayout>
+      <Avatar color="theme"  >Theme  </Avatar>
```
Comment (brentertz) [inline src/website/app/demos/Avatar/examples/colors.js]: Again to consider not having a "theme" color name, but rather having it default to the theme color.

File: src/website/app/demos/Avatar/examples/sizes.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+  source: `
+    <DemoLayout>
+      <Avatar size="small">
+        <img src="/images/Avatar.svg" alt="Sam" />
+      </Avatar>
+      <Avatar size="small">Sam</Avatar>
+      <Avatar size="small">
+        <IconCloud title="cloud" />
+      </Avatar>
+      <br />
```
Comment (brentertz) [inline src/website/app/demos/Avatar/examples/sizes.js]: Suggest to add line breaks following the `br` tags for readability.

File: src/website/app/demos/Avatar/examples/sizes.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+      <Avatar size="jumbo">
+        <img src="/images/Avatar.svg" alt="Jennifer" />
+      </Avatar>
+      <Avatar size="jumbo">Jennifer</Avatar>
+      <Avatar size="jumbo">
+        <IconCloud title="cloud" />
+      </Avatar>
+      <br />
```
Comment (brentertz) [inline src/website/app/demos/Avatar/examples/sizes.js]: Prob don't need this last `br`.

File: src/website/app/demos/Card/examples/card-title/rtl.js [hunk 14]
```diff
@@ -41,7 +43,7 @@ variable is set to \`rtl\` (right-to-left).`,
         <DemoLayout>
           <Card>
             <CardTitle
-              avatar={<img src="/images/215x210_avatar.png" alt="نص بديل" />}
+              avatar={<Avatar><img src="/images/Avatar.svg" alt="نص بديل" /></Avatar>}
```
Comment (brentertz) [inline src/website/app/demos/Card/examples/card-title/rtl.js]: Should prolly lowercase `avatar.svg`.

File: src/Avatar/Avatar.js [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+
+    return {
+      alignItems: 'center',
+      backgroundColor: props.noBackground
+        ? null
+        : props.color === 'theme'
+          ? theme.color_theme_60
+          : colors[`${props.color}_60`],
+      color,
+      borderRadius: '100%',
```
Comment (brentertz) [inline src/Avatar/Avatar.js]: Maybe we add a `shape` prop to allow the user to render the Avatar as a square or circle, with the default being circle.  @coldpour may also want to render a rhombus, but I don't think we should support that.

File: src/website/app/demos/Avatar/examples/customTextAbbr.js (hunk 1 above)
Comment (kylegach) [reply src/website/app/demos/Avatar/examples/customTextAbbr.js]: Thanks; addressed.

File: src/website/app/demos/Card/examples/card-status/importSyntax.js (hunk 5 above)
Comment (kylegach) [reply src/website/app/demos/Card/examples/card-status/importSyntax.js]: Oops. Thanks for the catch.

File: src/Avatar/Avatar.js (hunk 6 above)
Comment (kylegach) [reply src/Avatar/Avatar.js]: Done.

File: src/Avatar/Avatar.js (hunk 7 above)
Comment (kylegach) [reply src/Avatar/Avatar.js]: Nope.

File: src/Avatar/Avatar.js (hunk 8 above)
Comment (kylegach) [reply src/Avatar/Avatar.js]: Good idea. I also added the ability to use any color for both background and text.

File: src/Avatar/Avatar.js (hunk 10 above)
Comment (kylegach) [reply src/Avatar/Avatar.js]: TIL...

File: src/website/app/demos/Avatar/examples/sizes.js (hunk 12 above)
Comment (kylegach) [reply src/website/app/demos/Avatar/examples/sizes.js]: Done.

File: src/website/app/demos/Card/examples/card-title/rtl.js (hunk 14 above)
Comment (kylegach) [reply src/website/app/demos/Card/examples/card-title/rtl.js]: Done throughout.

File: src/Avatar/Avatar.js (hunk 15 above)
Comment (kylegach) [reply src/Avatar/Avatar.js]: https://491-card-supplementals--mineral-ui.netlify.com/components/avatar/shapes/

In addition to the default, circle, I added square and rounded.

File: src/colors/blue.js [hunk 16]
```diff
@@ -25,5 +25,7 @@ export default {
   blue_70: '#1f5dc2',
   blue_80: '#164ea8',
   blue_90: '#114091',
-  blue_100: '#0f397d'
+  blue_100: '#0f397d',
+
+  a11y_text_light: [60, 70, 80, 90, 100]
```
Comment (coldpour) [inline src/colors/blue.js]: This should still be a Set. IE11 has the add method just not the Array constructor. We can polyfill or use reduce

[1, 2, 3].reduce(function(acc, v) {acc.add(v); return acc;}, new Set())

File: src/colors/getReadableTextColor.js [hunk 17]
```diff
@@ -0,0 +1,11 @@
+import keyedColors from './keyedColors';
+import common from './common';
+
+export default (key: string, level: number) => {
+  if (keyedColors[key]) {
+    return keyedColors[key].a11y_text_light.indexOf(level) != -1
```
Comment (coldpour) [inline src/colors/getReadableTextColor.js]: Array.indexOf should be Set.has

File: src/colors/blue.js (hunk 16 above)
Comment (kylegach) [reply src/colors/blue.js]: We may optimize this at a later date, but shipping as-is for now.

**Comment by trotzig** [comment]
The missing avatar in the happo diff is a bug with the happo.io client. I've pushed a fix and will release a new version shortly.

**Comment by trotzig** [comment]
I've pushed https://github.com/mineral-ui/mineral-ui/pull/537 to address. Thanks for helping me find these edge-cases!

**Comment by trotzig** [comment]
@kylegach Not sure why internet explorer doesn't render the svg properly. It might be something in the happo pipeline, can you check if your asset renders nicely in IE11?

**Comment by kylegach** [comment]
@trotzig — In IE11, that image renders as an orange circle (no texture and no white circles on top). I suspect this is because the svg itself is using some fancy `mix-blend-mode` stuff.

**Pull request merged**